sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/ContinuousExecution.scala

```diff
@@ -266,6 +263,16 @@ class ContinuousExecution(
         SQLExecution.withNewExecutionId(
           sparkSessionForQuery, lastExecution)(lastExecution.toRdd)
       }
+    } catch {
+      case parentException: SparkException if parentException.getCause != null =>
```

Comment:
Can you add comments on what this code segment is doing?

Reply:
done